From 1f27018da93bb3f4a1a6ff17c6c3e5897f09265b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A1t=C3=A9=20Kukri?= <kukri.mate@gmail.com>
Date: Wed, 17 Apr 2019 14:00:58 +0200
Subject: [PATCH 2/3] Kill sys-include

---
 gcc/configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/configure b/gcc/configure
index 6121e1632..34de2d82e 100755
--- a/gcc/configure
+++ b/gcc/configure
@@ -3556,7 +3556,7 @@ else
 
  TARGET_SYSTEM_ROOT=
  TARGET_SYSTEM_ROOT_DEFINE=
- CROSS_SYSTEM_HEADER_DIR='$(gcc_tooldir)/sys-include'
+ CROSS_SYSTEM_HEADER_DIR='$(gcc_tooldir)/include'
 
 fi
 
-- 
2.21.0

